screen-orientation/onchange-event.html

```diff
@@ -6,10 +6,14 @@
   const type = screen.orientation.type;
   // wrap onchange listener in new promise to catch unreached assertion
   return new Promise(async (resolve, reject) => {
-    screen.orientation.onchange = () => {
-      reject(new Error("change event should not be fired"));
+    try {
```

Comment:
This is not necessary with `promise_test`, right?

Reply:
`promise_test` could not catch error in `new Promise()`, that's why test timeout if `orientation.lock` is unsupported, with `try..catch` we can reject the error to `promise_test`, you reminded me that the assertion should also be enclosed in `try...catch`

Reply:
It...should work. This fails the test for me:
```
<!doctype html>
<script src=/resources/testharness.js></script>
<script src=/resources/testharnessreport.js></script>
<script>
  promise_test(() => {
    return new Promise(resolve => {
      something.something.something;
    })
  })
</script>
```

Reply:
Oh I see, it's because of `await`. Hmm....

Reply:
You can do 
```
await t.step(() => screen.orientation.lock(type));
```

Reply:
Would this work?

```js
promise_test(async t => {
  const type = screen.orientation.type;
  screen.orientation.onchange = t.unreached_func("change event should not be fired");
  // wrap onchange listener in new promise to catch unreached assertion
  await screen.orientation.lock(type);
  assert_equals(screen.orientation.type, type);
  await new Promise(resolve => setTimeout(resolve));
}, "Test that orientationchange event is not fired when the orientation does not change.");
```

Reply:
@kaixinjxq can you try @Ms2ger's suggestion?

Reply:
Good suggestion, it's simplified and effective, but it's back to the discussion about using `SetTimeout` at https://github.com/web-platform-tests/wpt/pull/10701#discussion_r188504124, @marcoscaceres

Reply:
My suggestion without knowing the API at all would be to keep the zero-timeout `setTimeout` call (though it probably should use `t.step_timeout()` instead, and remove the entry from `lint.whitelist`).

If people are concerned about that, though, just drop the `await new Promise(resolve => setTimeout(resolve));` line and keep the other changes? The current code is very hard to read.